src/map.cpp

```diff
@@ -6910,6 +6910,112 @@ void map::restock_fruits( const tripoint &p, int time_since_last_actualize )
     }
 }
 
+void map::produce_sap( const tripoint &p, int time_since_last_actualize )
+{
+    if( time_since_last_actualize <= 0 ) {
+        return;
+    }
+
+    if( ter_at( p ).id != "t_tree_maple_tapped" ) {
```

Comment:
Minor: You could speed this check up (since it will be checked for ALL loaded tiles, even those on different z-levels) by comparing the int id rather than the string one. I think it's `ter( p ) != t_tree_maple_tapped`, because as far as I recall, you declared the int id for tapped trees in mapdata.cpp.